__tests__/app.test.js

```diff
@@ -79,3 +79,60 @@ describe("GET Requests", () => {
     });
   });
 });
+
+describe("PATCH Requests", () => {
+  describe("/api/articles/:article_id", () => {
+    test("status 200:updates given article and responds with the updated article (positive number)", () => {
+      const newObj = { inc_votes: 1 };
+      return request(app)
+        .patch("/api/articles/1")
+        .send(newObj)
+        .expect(200)
+        .then(({ body }) => {
+          const { article } = body;
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: 1,
+              body: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: 101,
+            })
+          );
+        });
+    });
+    test("status 200:updates given article and responds with the updated article (negative number)", () => {
+      const newObj = { inc_votes: -100 };
+      return request(app)
+        .patch("/api/articles/1")
+        .send(newObj)
+        .expect(200)
+        .then(({ body }) => {
+          const { article } = body;
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: 1,
+              body: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: 0,
+            })
+          );
+        });
+    });
+    test("if passed a number that will take the total to less than 0, throws an error", () => {
```

Comment:
What if the id given is invalid, or doesn't exist in the database? What if the inc_votes value isn't a number? What if there's no inc_votes value at all? 😱

Reply:
i had a test in for if inc_votes is passed something that isnt a number but when i did the pull request i got told something like we dont need that because of happy path